From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ethan Hill <ethan.hill1999@outlook.com>
Date: Wed, 21 Oct 2026 15:59:11 -0500
Subject: [PATCH 2/2] KITKAT: KitKat UI mods - P9 navbar, P10 completion sweep
 (framework theme/fonts, SystemUI shade/QS/recents/ticker/navbar, Settings
 icons, hybrid 2013 emoji font, easter egg)

---
 res/values/styles.xml |  5 +++++
 res/values/themes.xml | 29 +++++++++++++++++------------
 2 files changed, 22 insertions(+), 12 deletions(-)

diff --git a/res/values/styles.xml b/res/values/styles.xml
index 0a28b016ca9..5bb3ce59f56 100644
--- a/res/values/styles.xml
+++ b/res/values/styles.xml
@@ -758,7 +758,12 @@
         <item name="android:gravity">start</item>
     </style>
 
+    <!-- KitKat UI Project (11.6): the huge CollapsingToolbarTitle.Expanded
+         "Settings" header is a modern signature; 4.4 carried a modest action
+         bar title. Sized down to a compact 20sp primary title. -->
     <style name="HomepageTitleText" parent="CollapsingToolbarTitle.Expanded">
+        <item name="android:textSize">20sp</item>
+        <item name="android:textColor">?android:attr/textColorPrimary</item>
         <item name="android:layout_width">wrap_content</item>
         <item name="android:layout_height">wrap_content</item>
         <item name="android:layout_marginBottom">@dimen/homepage_title_margin_bottom</item>
diff --git a/res/values/themes.xml b/res/values/themes.xml
index b149bb8fd6b..1de867cb9bb 100644
--- a/res/values/themes.xml
+++ b/res/values/themes.xml
@@ -61,7 +61,7 @@
 
         <!-- For AppBarLayout -->
         <item name="colorPrimary">@*android:color/primary_device_default_settings_light</item>
-        <item name="colorPrimaryVariant">@android:color/white</item>
+        <item name="colorPrimaryVariant">@*android:color/primary_dark_device_default_settings</item>
 
         <!-- For slice view in settings -->
         <item name="sliceViewStyle">@style/Widget.SliceView.Settings</item>
@@ -86,8 +86,11 @@
         <!-- Redefine the ActionBar style for contentInsetStart -->
         <item name="android:actionBarStyle">@style/Widget.ActionBar.SubSettings</item>
 
-        <item name="android:navigationBarColor">@android:color/transparent</item>
-        <item name="android:statusBarColor">@android:color/transparent</item>
+        <!-- KitKat UI Project (11.6): 4.4 subpage bars are solid black
+             (transparent bars are an edge-to-edge-era look; with the P2
+             OPAQUE bar default these must be pinned black). -->
+        <item name="android:navigationBarColor">@android:color/black</item>
+        <item name="android:statusBarColor">@android:color/black</item>
     </style>
 
     <style name="Theme.SubSettings" parent="Theme.SubSettings.Base"/>
@@ -183,24 +186,26 @@
     </style>
 
     <style name="Theme.Settings.HomeBase" parent="Theme.Settings.NoActionBar">
-        <item name="colorPrimary">@*android:color/primary_device_default_settings_light</item>
-        <item name="colorAccent">@*android:color/accent_device_default_light</item>
+        <!-- KitKat UI Project (11.6): homepage palette forced to the
+             DeviceDefault DARK pins (4.4 Settings was dark). -->
+        <item name="colorPrimary">@*android:color/primary_dark_device_default_settings</item>
+        <item name="colorAccent">@*android:color/accent_device_default_dark</item>
         <item name="preferenceTheme">@style/SettingsPreferenceTheme</item>
 
         <!-- action bar, needed for search bar icon tinting -->
         <item name="android:actionBarTheme">@*android:style/ThemeOverlay.DeviceDefault.ActionBar</item>
 
         <!-- Define this color for material design -->
-        <item name="colorPrimaryVariant">@android:color/white</item>
+        <item name="colorPrimaryVariant">@*android:color/primary_dark_device_default_settings</item>
     </style>
 
     <style name="Theme.Settings.Home" parent="Theme.Settings.HomeBase">
-        <item name="colorPrimaryDark">@*android:color/primary_dark_device_default_settings_light</item>
-        <item name="android:windowLightStatusBar">true</item>
-        <item name="android:navigationBarDividerColor">@*android:color/ripple_material_light</item>
-        <!-- Homepage should follow device default design, the values is same as device default theme.-->
-        <item name="android:navigationBarColor">@android:color/white</item>
-        <item name="android:statusBarColor">?attr/colorPrimaryDark</item>
+        <!-- KitKat UI Project (11.6): 4.4 Settings homepage - dark bars,
+             dark status bar (no light status bar flag), no nav divider. -->
+        <item name="colorPrimaryDark">@*android:color/primary_dark_device_default_settings</item>
+        <item name="android:windowLightStatusBar">false</item>
+        <item name="android:navigationBarColor">@android:color/black</item>
+        <item name="android:statusBarColor">@android:color/black</item>
     </style>
 
     <style name="Theme.Settings.Home.NoAnimation">
-- 
2.53.0

